### PR TITLE
Allow annotation instruments base on task type

### PR DESCRIPTION
**Pull request by Dzeranov**
### Motivation and context

Originally, in CVAT there were no restrictions about annotation tools usage. It can mislead users and doesn't guarantee an annotation results consistency.
3 different task modes were added: Tag, Rectangle and Polyshape.
For Tag only tag annotation is allowed.
For Rectangle drawing new rectangle is allowed.
For Polyshape almost every tool is allowed expect tag annotation and rectangle.   

```python
# Copyright (C) 2021 Intel Corporation
#
# SPDX-License-Identifier: MIT
```

### CODE REVIEW

File: cvat/apps/engine/serializers.py [hunk 1]
```diff
@@ -391,7 +391,7 @@ class Meta:
             'bug_tracker', 'created_date', 'updated_date', 'overlap',
             'segment_size', 'status', 'labels', 'segments',
             'data_chunk_size', 'data_compressed_chunk_type', 'data_original_chunk_type', 'size', 'image_quality',
-            'data', 'dimension', 'subset')
+            'data', 'dimension', 'subset', 'allowed_annotation_mode')
```
Comment (vkomodey) [inline cvat/apps/engine/serializers.py:394]: `allowed_annotation_instrument` maybe?

Comment (vkomodey) [reply cvat/apps/engine/serializers.py:394]: am I right, that for each job type there is always only one instrument?

File: cvat-ui/src/components/annotation-page/top-bar/right-group.tsx [hunk 2]
```diff
@@ -185,9 +185,19 @@ function RightGroup(props: Props): JSX.Element {
...
+                            return (
+                                <Select.Option disabled={ws === Workspace.ATTRIBUTE_ANNOTATION} key={ws} value={ws}>
+                                    {ws}
+                                </Select.Option>
+                            );
+                        }
                         if (ws !== Workspace.STANDARD3D) {
                             return (
-                                <Select.Option key={ws} value={ws}>
+                                <Select.Option disabled={ws !== Workspace.STANDARD} key={ws} value={ws}>
```
Comment (vkomodey) [inline cvat-ui/src/components/annotation-page/top-bar/right-group.tsx:200]: you are already in the if branch, where `ws !== Workspace.STANDARD3D`. this condition is always true here. just leave
```
<Select.Option disabled key={ws} value={ws}>
```

**Review by vkomodey — CHANGES_REQUESTED** [review]

File: cvat/apps/engine/serializers.py (hunk 1 above)
Comment (Dzeranov) [reply cvat/apps/engine/serializers.py:394]: Not one instrument but one group of instruments. But actually, Rectangle is being used for drawing bounding boxes, Tag is for Tag annotation (binary or multiple choise),
Polyshape for drawing polygons (Polylines, polygons etc.).

File: cvat-ui/src/components/annotation-page/top-bar/right-group.tsx (hunk 2 above)
Comment (Dzeranov) [reply cvat-ui/src/components/annotation-page/top-bar/right-group.tsx:200]: Are you sure?
`if (ws !== Workspace.STANDARD3D)` means that there are 3 possible variants returning true.
And `disabled={ws !== Workspace.STANDARD}` makes them disabled except of STANDARD.

Comment (vkomodey) [reply cvat-ui/src/components/annotation-page/top-bar/right-group.tsx:200]: ahh, ok. agree

**Pull request merged**